b2b-app/imports/ui/admin/register/context.js

```diff
@@ -0,0 +1,104 @@
+import React, { useReducer, createContext } from 'react'
+import PropTypes from 'prop-types'
+
+import useMediaQuery from '@material-ui/core/useMediaQuery'
+import { useTheme } from '@material-ui/core/styles'
+
+const steps = ['Contact details', 'About you', 'Emergency contact', 'Avatar', 'Confirm']
+
+const initialState = {
+  models: Array(steps.length)
+    .fill()
+    .map(() => ({})),
+  ui: {
+    activeStep: 0,
+    isEditingStep: false,
+    isSubmitting: false,
+  },
+}
+
+const reducer = (state, action) => {
+  const {
+    models,
+    ui: { activeStep },
+  } = state
+  switch (action.type) {
+    case 'go_next':
+      return {
+        models: models.map((model, idx) => (idx === activeStep ? action.model : model)),
```

Comment:
should we have a standard convention to specify index as a variable? I usually use `i`

Reply:
I like `i`, `j` as well for integer loop variables, or even `ix`
Generally I like shorter variable names and file names, It saves typing, reduces mistakes. Quite often something simple like `save` is better than `saveMyDataToTheDatabase`
Avoiding variables completely is better, but can't be completely avoided